tests/test/unify.rs

```diff
@@ -78,6 +78,34 @@ fn forall_equality() {
             InEnvironment { environment: Env([]), goal: '!2_0: '!2_1 }, \
             InEnvironment { environment: Env([]), goal: '!2_1: '!2_0 }]"
         }
+
+        goal {
+            // Function pointers with different ABIs should not be equal.
+            for<'a, 'b> extern "Rust" fn(Ref<'a, Ref<'b, Unit>>): Eq<for<'c, 'd> extern "C" fn(Ref<'c, Ref<'d, Unit>>)>
+        } yields {
+            "No possible solution"
+        }
+
+        goal {
+            // Function pointers with identical ABIs should be equal.
+            for<'a, 'b> extern "Rust" fn(Ref<'a, Ref<'b, Unit>>): Eq<for<'c, 'd> extern "Rust" fn(Ref<'c, Ref<'d, Unit>>)>
+        } yields {
+            "Unique; substitution [], lifetime constraints []"
+        }
+
+        goal {
+            // Function pointers with different safety should not be equal.
+            for<'a, 'b> unsafe fn(Ref<'a, Ref<'b, Unit>>): Eq<for<'c, 'd> fn(Ref<'c, Ref<'d, Unit>>)>
```

Comment:
it feels like this test is overly complex for what it is testing. Maybe `unsafe fn(): Eq<fn()>` would be simpler?